lsp/nls/src/cache.rs

```diff
@@ -47,12 +47,35 @@ impl CacheExt for Cache {
         let mut import_errors = Vec::new();
         if let Ok(CacheOp::Done((ids, errors))) = self.resolve_imports(file_id) {
             import_errors = errors;
+            let mut errors = Vec::new();
             for id in ids {
-                self.typecheck_with_analysis(id, initial_ctxt, initial_env, lin_cache)
-                    .unwrap();
+                // Ignore the results, and check for errors after resolution
+                match self.typecheck_with_analysis(id, initial_ctxt, initial_env, lin_cache) {
+                    Ok(..) => {}
+                    Err(..) if !lin_cache.contains_key(&id) => {
+                        let name: String = self.name(id).to_str().unwrap().into();
+                        if let Some(pos) = lin_cache
+                            .get(&file_id)
+                            .and_then(|lin| lin.import_locations.get(&id))
+                        {
+                            errors.push((name, *pos));
+                        }
+                    }
+                    Err(..) => {}
+                }
             }
+            let message = "This import could not be resolved because its content have either failed to parse or typecheck correctly.";
+            let errors = errors
+                .into_iter()
+                .map(|(name, pos)| ImportError::IOError(name, String::from(message), pos));
+
+            import_errors.extend(errors);
         }
 
+        // Two things
```

Comment:
which code does this comment refer to?

Reply:
Ah, this is some noise from the draft implementation.  PTAL